packages/gestalt/src/Modal.js

```diff
@@ -95,6 +100,30 @@ export default function Modal({
     }
   };
 
+  const updateShadows = () => {
+    const target = content.current;
+    if (!target) {
+      return;
+    }
+    const hasVerticalScrollbar = target.clientHeight < target.scrollHeight;
+    setShowTopShadow(hasVerticalScrollbar && target.scrollTop > 0);
+    setShowBottomShadow(
+      hasVerticalScrollbar &&
+        target.offsetHeight + target.scrollTop < target.scrollHeight
+    );
+  };
+
+  React.useEffect(() => {
+    window.addEventListener('resize', updateShadows);
+    return () => {
+      window.removeEventListener('resize', updateShadows);
+    };
+  }, []);
+
+  React.useEffect(() => {
+    updateShadows();
+  }, []);
+
```

Comment:
This code makes me realize we don't have the hooks exhaustive deps linter plugged in properly. In this case, the `useEffect` calls should have `updateShadows` set as deps, and `updateShadows` should be memoized. I'm fairly certain there's no bugs in the way you've written it, but it'd be better to make it future-proof.

Reply:
@btraut interesting - we should have enabled it here:
https://github.com/pinterest/gestalt/blob/455e6d3bdc7caad0ca0991d692fb65219eea8353/.eslintrc.json#L61

adding `updateShadows` to the dependency list does show the follow expected lint error: 
![image](https://user-images.githubusercontent.com/127199/82264769-6a4f4200-991a-11ea-91c6-f4e49a641210.png)

For this particular line we only want to call `updateShadows` on initial render, so the the empty array should do the trick.